core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -155,21 +155,36 @@ public ClientLogger asError() {
      */
     public void log(String format, Object... args) {
         if (canLogAtLevel(level)) {
-            performLogging(format, args);
+            performLogging(format, true, args);
         }
     }
 
+    /**
+     * This will log, if error log level is enabled , and throw the runtime exception.
+     * @param runtimeException to be thrown. It will do nothing if {@code null} is provided.
+     */
+    public void logAndThrow(RuntimeException runtimeException) {
+        if(runtimeException == null) {
+            return;
+        }
+        // By default we are treating runtime exception as error ?
+        level = ERROR_LEVEL;
+        if (canLogAtLevel(level)) {
+            performLogging(runtimeException.getClass().getName(), false, runtimeException);
+        }
+        throw runtimeException;
+    }
+
     /*
      * Performs the logging.
      * @param format Format-able message.
      * @param args Arguments for the message, if an exception is being logged last argument is the throwable.
      */
-    private void performLogging(String format, Object... args) {
+    private void performLogging(String format, boolean removeThroawable, Object... args) {
         // If the logging level is less granular than verbose remove the potential throwable from the args.
-        if (configurationLevel > VERBOSE_LEVEL) {
+        if (removeThroawable && configurationLevel > VERBOSE_LEVEL) {
             args = attemptToRemoveThrowable(args);
         }
```

Comment:
We do want to log Runtime Exception  if someone has called logAndThrow().   But we want to remove Runtime Exception if some one is calling regular log() ….  Thus we need Boolean flag, removeThroawable,  for this flipping